ris.go

```diff
@@ -0,0 +1,154 @@
+	PeerASN   string      `json:"peer_asn,omitempty"`
+	ID        string      `json:"id,omitempty"`
+	Raw       string      `json:"raw,omitempty"`
+	Host      string      `json:"host,omitempty"`
+}
+
+// RIS defines RIS message format
+type RIS struct {
+	Type string   `json:"type,omitempty"`
+	Data *RISData `json:"data,omitempty"`
+}
+
+type Message struct {
+	CommonHeader  *bmp.CommonHeader
+	PerPeerHeader *bmp.PerPeerHeader
+}
+
+func main() {
+	flag.Parse()
+	_ = flag.Set("logtostderr", "true")
+	ris, err := http.Get(stream)
+	if err != nil {
+		glog.Errorf("failed to connect to RIS source with error: %+v", err)
+		os.Exit(1)
+	}
+	defer ris.Body.Close()
+	bmpSrv, err := net.Dial("tcp", bmpAddress)
+	if err != nil {
+		glog.Errorf("failed to connect to destination with error: %+v", err)
+		os.Exit(1)
+	}
+	defer bmpSrv.Close()
+	glog.Infof("connection to bmp %v established", bmpSrv.RemoteAddr())
+
+	reader := bufio.NewReader(ris.Body)
+	errorCh := make(chan error)
+	for {
+		m := &RIS{}
+		b, err := reader.ReadBytes('\n')
+		if err != nil {
+			glog.Errorf("failed to read message with error: %+v", err)
+			os.Exit(1)
+		}
+		go func(b []byte, errorCh chan error) {
+			if err := json.Unmarshal(b, m); err != nil {
+				glog.Errorf("failed to decode streamed message with error: %+v", err)
+				errorCh <- err
+				return
+			}
+			bmpMsg := Message{}
+			bmpMsg.CommonHeader = &bmp.CommonHeader{
+				Version:     3,
+				MessageType: bmp.RouteMonitorMsg,
+			}
+			bmpMsg.PerPeerHeader = &bmp.PerPeerHeader{
+				PeerType:          0, // *  Peer Type = 0: Global Instance Peer
+				PeerDistinguisher: make([]byte, 8),
+				PeerAddress:       make([]byte, 16),
+				PeerBGPID:         net.ParseIP(bgpID).To4(),
+				PeerTimestamp:     make([]byte, 8),
+			}
+			// Populating peer address
+			pa := net.ParseIP(m.Data.Peer)
+			if pa.To4() != nil {
+				bmpMsg.PerPeerHeader.FlagV = false
+				copy(bmpMsg.PerPeerHeader.PeerAddress[12:16], pa.To4())
+			} else if pa.To16() != nil {
+				bmpMsg.PerPeerHeader.FlagV = true
+				copy(bmpMsg.PerPeerHeader.PeerAddress, pa.To16())
+			} else {
+				glog.Warningf("invalid peer address %s", m.Data.Peer)
+				errorCh <- err
+				return
+			}
+			// Populating Peer ASN
+			asn, err := strconv.Atoi(m.Data.PeerASN)
+			if err != nil {
+				glog.Warningf("invalid peer asn %s", m.Data.PeerASN)
+			}
+			bmpMsg.PerPeerHeader.PeerAS = int32(asn)
+
+			// Populating peer timestamp
+			t := strings.Split(m.Data.Timestamp.String(), ".")
+			sec, _ := strconv.Atoi(t[0])
+			msec := 0
+			if len(t) > 1 {
+				msec, _ = strconv.Atoi(t[1])
+			}
+			binary.BigEndian.PutUint32(bmpMsg.PerPeerHeader.PeerTimestamp[0:4], uint32(sec))
+			binary.BigEndian.PutUint32(bmpMsg.PerPeerHeader.PeerTimestamp[4:8], uint32(msec))
+
+			raw, err := hex.DecodeString(m.Data.Raw)
+			if err != nil {
+				glog.Warningf("invalid raw data, failed to decode with error: %+v", err)
+			}
+
+			bmpMsg.CommonHeader.MessageLength = int32(6 + 42 + len(raw))
```

Comment:
instead of 6 and 42 please use meaningful constant. applies to all lines where numbers are used instead of size or offset. thanks

Reply:
approving for now, can make change through new PR